src/frontend/apps/desk/src/features/mail-domains/components/MailDomainsContent.tsx

```diff
@@ -1,51 +1,101 @@
 import { useTranslation } from 'react-i18next';
 
-import { Card } from '@/components';
+import { Box, Card, Text, TextErrors } from '@/components';
+import { MailDomain } from '@/features/mail-domains';
+import { useMailDomain } from '@/features/mail-domains/api/useMailDomain';
+import { useMailDomainMailboxes } from '@/features/mail-domains/api/useMailDomainMailboxes';
+import { PAGE_SIZE } from '@/features/mail-domains/conf';
 
-export function MailDomainsContent() {
+import { default as MailDomainsLogo } from '../assets/mail-domains-logo.svg';
+export type ViewMailbox = { email: string; id: string };
+
+export function MailDomainsContent({ mailDomain }: { mailDomain: MailDomain }) {
   const { t } = useTranslation();
+  const pagination = usePagination({
+    defaultPage: 1,
+    pageSize: PAGE_SIZE,
+  });
+  const { page, pageSize, setPagesCount } = pagination;
 
-  const dataset = [
-    {
-      id: '1',
-      name: 'John Doe',
-      email: 'john@doe.com',
-      state: 'Active',
-      lastConnection: '2021-09-01',
-    },
-    {
-      id: '2',
-      name: 'Jane Doe',
-      email: 'jane@doe.com',
-      state: 'Inactive',
-      lastConnection: '2021-09-02',
-    },
-  ];
+  const { data, isLoading, error } = useMailDomainMailboxes({
+    id: mailDomain.id,
+    page,
+  });
 
-  return (
-    <Card $padding="small" $margin="large">
-      <DataGrid
-        columns={[
-          {
-            headerName: t('Names'),
-            field: 'name',
-          },
-          {
-            field: 'email',
-            headerName: t('Emails'),
-          },
-          {
-            field: 'state',
-            headerName: t('State'),
-          },
-          {
-            field: 'lastConnection',
-            headerName: t('Last Connecttion'),
-          },
-        ]}
-        rows={dataset}
-      />
-    </Card>
+  const viewMailboxes: ViewMailbox[] =
+    mailDomain && data?.results?.length
+      ? data.results.map((mailbox) => ({
+          email: `${mailbox.local_part}@${mailDomain.name}`,
+          id: mailbox.id,
+        }))
+      : [];
+
+  useEffect(() => {
+    setPagesCount(data?.count ? Math.ceil(data.count / pageSize) : 0);
+  }, [data?.count, pageSize, setPagesCount]);
+  return isLoading ? (
+    <Box $align="center" $justify="center" $height="100%">
+      <Loader />
+    </Box>
+  ) : (
+    <>
+      <TopBanner mailDomainId={mailDomain.id} />
+      <Card
+        $padding={{ bottom: 'small' }}
+        $margin={{ all: 'big', top: 'none' }}
+        $overflow="auto"
+      >
+        {error && <TextErrors causes={error.cause} />}
+        <DataGrid
+          columns={[
+            {
+              field: 'email',
+              headerName: t('Emails'),
+            },
+          ]}
+          rows={viewMailboxes}
+          isLoading={isLoading}
+          pagination={{
+            ...pagination,
+            displayGoto: false,
+          }}
+          aria-label={t('Mailboxes list')}
```

Comment:
Cunningham display a warning about the sorting implementation. 
You have a implementation exemple here:
https://github.com/numerique-gouv/people/blob/d2c7eaaa4b1f9053a6a8ad5dc96233c061bf7236/src/frontend/apps/desk/src/features/members/components/MemberGrid.tsx#L180